module.nix

```diff
@@ -171,9 +195,9 @@ in
             # TODO maybe make this configurable
             ExecStart = "${cfg.authentikComponents.celery}/bin/celery -A authentik.root.celery worker -Ofair --max-tasks-per-child=1 --autoscale 3,1 -E -B -s /tmp/celerybeat-schedule -Q authentik,authentik_scheduled,authentik_events";
             EnvironmentFile = mkIf (cfg.environmentFile != null) [ cfg.environmentFile ];
-            LoadCredential = mkIf (cfg.nginx.enable && cfg.nginx.enableACME) [
-              "${cfg.nginx.host}.pem:${config.security.acme.certs.${cfg.nginx.host}.directory}/fullchain.pem"
-              "${cfg.nginx.host}.key:${config.security.acme.certs.${cfg.nginx.host}.directory}/key.pem"
+            LoadCredential = mkIf (cfg.nginx.enable && cfg.nginx.vhost.enableACME) [
```

Comment:
What happens on evaluation if you actually set `cfg.nginx.vhost.enableACME = true;`?
Additionally, what happens on evaluation if you actually set `cfg.nginx.vhost = {enableACME = true; useACMEHost = "your.host";}`?

What troubles me is, these are changes which, if merged like that, will make the service factually unusable (due to evaluation errors) for others who are currently using `enableACME`.
It is obvious to observe, you didn't evaluate the config with this setting and you didn't check for the effectiveness of the change you introduced here either.
Please make sure to check the changes you introduce really do work as you intend them to.

Reply:
Hi thanks for the feedback, although I don't quite understand it.

My current deployment where I tested the change is using the following configuration.

```nix
{ inputs, config, ... }:
{
  imports = [ inputs.authentik.nixosModules.default ];

  sops.secrets.authentik.owner = "root";

  services.authentik = {
    enable = true;

    environmentFile = config.sops.secrets.authentik.path;
    postgres.enable = true;

    settings = {
      avatars = "initials";
      default_user_change_email = true;
    };

    nginx = {
      enable = true;
      host = "auth.m4rc3l.de";
      vhost = {
        forceSSL = true;
        useACMEHost = "m4rc3l.de";
        listen = [{
          addr = "0.0.0.0";
          port = 8443;
          ssl = true;
          proxyProtocol = true;
        }];
      };
    };
  };
}
```

And it works without problems.

Reply:
Ohhh nvm, now I see what you mean. Setting and custom acme host in conjunction with enableACME. I'll test this out and try to fix all errors. Thanks for that hint.

Reply:
This was a theoretical argument to make clear that you'll have to set `enableACME` together with `useACMEHost` to get the place where you introduced the use of `useACMEHost`.
While really it is even asserted to be impossible in nginx module to have both settings set at the same time.

Reply:
I've just noticed it. It worked so I did not question it. But it seems like authentik does not need certificates in the first place. I'll look into it.

Reply:
It turns out authentik does not need the certificates unconditionally, because it generates a self-signed certificate on the first start (https://goauthentik.io/docs/core/certificates), **however** this self-signed certificate will only be valid for 1 year.
Since the `proxy_pass` directive further down points to a secured server it might stop working when the self-signed certificate is outdated.
The certificates passed through to authentik via `LoadCredential` -> `env://CREDENTIALS_DIRECTORY` -> `cert_discovery_dir` will show up in authentik on https://auth.example.irg/if/admin/#/crypto/certificates and can be assigned to an tenant in https://auth.example.org/if/admin/#/core/tenants within `Other global settings`.

Reply:
Ok, another change that I would have made in the future that I've also already rested locally is changing the proxy pass to the http port 9000 instead of the https one 9443. Do you prehabs know the reason that currently the https variant is used.

Reply:
No.

Reply:
Back to the topic, right now the module is unusable for me with `enableACME` setting because it errors on evaluation.

Reply:
Thank you. This works for me with your latest changes.